packages/synapse-react-client/src/components/SetAccessRequirementCommonFields/SetAccessRequirementCommonFields.tsx

```diff
@@ -0,0 +1,327 @@
+import {
+  Alert,
+  Stack,
+  TextField,
+  Tooltip,
+  Typography,
+  TypographyProps,
+} from '@mui/material'
+import {
+  ACCESS_REQUIREMENT_CONCRETE_TYPE,
+  ACCESS_TYPE,
+  AccessRequirement,
+  MANAGED_ACT_ACCESS_REQUIREMENT_CONCRETE_TYPE_VALUE,
+  RestrictableObjectDescriptor,
+  RestrictableObjectType,
+  SELF_SIGN_ACCESS_REQUIREMENT_CONCRETE_TYPE_VALUE,
+} from '@sage-bionetworks/synapse-types'
+import pluralize from 'pluralize'
+import React, { useEffect, useImperativeHandle, useMemo, useState } from 'react'
+import {
+  useCreateAccessRequirement,
+  useGetAccessRequirements,
+  useUpdateAccessRequirement,
+} from '../../synapse-queries'
+import { SynapseClientError } from '../../utils/SynapseClientError'
+import EntitySubjectsSelector from '../EntitySubjectsSelector'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import TeamSubjectsSelector from '../TeamSubjectsSelector'
+import { RadioGroup } from '../widgets/RadioGroup'
+
+export const EMPTY_SUBJECT_LIST_ERROR_MESSAGE =
+  'Please select at least one resource for this Access Requirement to be associated with.'
+export const UNSAVED_SUBJECTS_ERROR_MESSAGE = (
+  subjectsType: RestrictableObjectType,
+) => {
+  const idsText =
+    subjectsType === RestrictableObjectType.ENTITY ? 'Synapse' : 'Team'
+  const typeText =
+    subjectsType[0].toUpperCase() + subjectsType.slice(1).toLowerCase()
+  return `${idsText} IDs were specified but not added to the subjects list. Please either clear out the Add ${idsText} IDs textbox or click the Add ${pluralize(
+    typeText,
+  )} button.`
+}
+export const MISSING_NAME_ERROR_MESSAGE =
+  'Please enter a name for this Access Requirement.'
+const NAME_HELP_TEXT =
+  "Enter access requirement name. This will also be used when sending notifications for expiring or revoked approval. For example, 'The approval for the name access requirement was revoked...'"
+
+const headerProps: Partial<TypographyProps> = {
+  variant: 'body1',
+  fontWeight: 700,
+}
+
+function getAccessType(subject: RestrictableObjectDescriptor) {
+  switch (subject.type) {
+    case RestrictableObjectType.ENTITY:
+      return ACCESS_TYPE.DOWNLOAD
+    case RestrictableObjectType.TEAM:
+      return ACCESS_TYPE.PARTICIPATE
+    default:
+      throw new Error(
+        `RestrictableObjectType ${subject.type} does not have an access type specified.`,
+      )
+  }
+}
+
+export type SetAccessRequirementCommonFieldsHandle = {
+  /* Allow the parent component to trigger saving the AR, so this may be embedded in an arbitrary modal. */
+  save: () => void
+}
+
+export type SetAccessRequirementCommonFieldsProps = {
+  /* Provided when creating a new AR */
+  subject?: RestrictableObjectDescriptor
+  /* Provided when editing an existing AR */
+  accessRequirementId?: string
+  /* Called when AR has been saved successfully */
+  onSave: (
+    accessRequirementId: string,
+    accessRequirementConreteType: ACCESS_REQUIREMENT_CONCRETE_TYPE,
+  ) => void
```

Comment:
When a `subject` is provided, then this component will create a new AR. `onSave` communicates the new AR's id and concrete type back to the parent (the AR wizard).